abdiff/core/build_ab_images.py

```diff
@@ -1 +1,97 @@
-"""abdiff.core.build_ab_images"""
+import logging
+import tempfile
+
+import docker
+import docker.models
+import docker.models.images
+from pygit2 import clone_repository
+from pygit2.enums import ResetMode
+
+from abdiff.core.utils import update_or_create_job_json
+
+logger = logging.getLogger(__name__)
+
+
+def build_ab_images(
+    job_directory: str,
+    commit_sha_a: str,
+    commit_sha_b: str,
+    docker_client: docker.client.DockerClient | None = None,
+) -> tuple[str, str]:
+    """Build Docker images based on 2 commit SHAs.
+
+    Args:
+        job_directory: The directory containing all files related to a job.
+        commit_sha_a: The SHA of the first commit for comparison.
+        commit_sha_b: The SHA of the second commit for comparison.
+        docker_client: A configured Docker client.
+    """
+    if not docker_client:
+        docker_client = docker.from_env()
+
+    image_tags = []
+    for commit_sha in [commit_sha_a, commit_sha_b]:
+        logger.debug(f"Processing commit: {commit_sha}")
+        image_tag = f"transmogrifier-{job_directory.split("/")[-1]}-{commit_sha}:latest"
+        if docker_image_exists(docker_client, image_tag):
+            logger.debug("Docker image already exists with tag: {image_tag}")
```

Comment:
Sorry to prompt another commit, but believe this needs an `f`-string prefix, just a normal string as-is.  This was the output in my local testing:

```
Docker image already exists with tag: {image_tag}
```

Reply:
Ugh, good catch!